haas/client/node.py

```diff
@@ -0,0 +1,121 @@
+import json
+from haas.client.base import ClientBase
+from haas.client import errors
+
+
+class Node(ClientBase):
+    """ Consists of calls to query and manipulate node related
+    objects and relations """
+
+    def list(self, is_free):
+        """ List all nodes that HIL manages """
+        url = self.object_url('nodes', is_free)
+        q = self.s.get(url)
+        if q.ok:
+            return q.json()
+        elif q.status_code == 401:
+            raise errors.AuthenticationError(
+                  "Make sure credentials match chosen authentication backend."
+                  )
```

Comment:
Cli is going to get redesigned so not worrying about it too much. As to the boiler plate code like above I agree about that. some errors look are common for all classes and should be inherited from the parent class. I do not have a good way of implementing it yet so left it like this for now. With recent additions to api and new api in the pipeline, I think I will let it be and refactor it at a later time when client library is relatively stable.